dtfs-central-api/api-tests/v1/tfm/deal-cancellation/tfm-deal-cancellation-post.api-test.ts

```diff
@@ -0,0 +1,156 @@
+          tfm: {
+            dateReceived: '23-09-2024',
+            dateReceivedTimestamp: 1727085149,
+            parties: {},
+            activities: [],
+            product: DEALS.DEAL_TYPE.BSS_EWCS,
+            stage: TFM_DEAL_STAGE.CONFIRMED,
+            exporterCreditRating: 'Acceptable (B+)',
+            lastUpdated: 1727085149571,
+            cancellation,
+            lossGivenDefault: 50,
+            probabilityOfDefault: 12,
+          },
+        },
+        auditDetails: generatePortalAuditDetails(MOCK_PORTAL_USER._id),
+      })
+      .to(`/v1/tfm/deals/${dealId}`);
+  });
+
+  afterEach(async () => {
+    process.env = { ...originalProcessEnv };
+    await wipeDB.wipe([MONGO_DB_COLLECTIONS.TFM_DEALS, MONGO_DB_COLLECTIONS.DEALS]);
+  });
+
+  describe('POST /v1/tfm/deals/:dealId/cancellation/submit', () => {
+    describe('when FF_TFM_DEAL_CANCELLATION_ENABLED is disabled', () => {
+      beforeEach(() => {
+        process.env.FF_TFM_DEAL_CANCELLATION_ENABLED = 'false';
+      });
+
+      afterAll(() => {
+        jest.resetAllMocks();
+      });
+
+      it('should return 404', async () => {
+        const { status } = await testApi.post({ cancellation, auditDetails }).to(submitDealCancellationUrl);
+
+        expect(status).toEqual(HttpStatusCode.NotFound);
+      });
+    });
+
+    describe('when FF_TFM_DEAL_CANCELLATION_ENABLED is enabled', () => {
+      beforeEach(() => {
+        process.env.FF_TFM_DEAL_CANCELLATION_ENABLED = 'true';
+      });
+
+      afterAll(() => {
+        jest.resetAllMocks();
+      });
+
+      withMongoIdPathParameterValidationTests({
+        baseUrl: '/v1/tfm/deals/:dealId/cancellation/submit',
+        makeRequest: (url) => testApi.post({ cancellation, auditDetails }).to(url),
+      });
+
+      it('should return the submit cancellation response object if a matching deal and cancellation exists', async () => {
+        const submitCancellationResponse = await testApi.post({ cancellation, auditDetails }).to(submitDealCancellationUrl);
+
+        expect(submitCancellationResponse.body).toEqual({ cancelledDeal: { ukefDealId: dealId } });
+        expect(submitCancellationResponse.status).toEqual(HttpStatusCode.Ok);
+      });
+
+      it('should return 404 if deal is valid but the existing cancellation does not match the passed in params', async () => {
+        const differentDealCancellationDetails = { ...cancellation, reason: 'a different reason' };
+
+        const submitCancellationResponse = await testApi.post({ cancellation: differentDealCancellationDetails, auditDetails }).to(submitDealCancellationUrl);
+
+        expect(submitCancellationResponse.status).toEqual(HttpStatusCode.NotFound);
+        expect(submitCancellationResponse.body).toEqual({
+          status: HttpStatusCode.NotFound,
+          message: `Deal not found: ${dealId}`,
+        });
+      });
+
+      it('should return 404 if dealId is valid but not associated to a record', async () => {
```

Comment:
Could we say deal instead of record please? I almost thought this was a "fee record" for a moment 😄

Reply:
Yep, good idea! Have changed here (and on the other deal cancellation tests where it was worded similarly)